packages/ui-components/package.json

```diff
@@ -0,0 +1,56 @@
+{
+  "name": "@cockroachlabs/ui-components",
+  "version": "0.1.0-alpha.20200207",
+  "description": "UI Components is a reusable component library. It provides the core components for the Cockroach design system.",
+  "repository": {
+    "type": "git",
+    "url": "https://github.com/cockroachdb/ui"
+  },
+  "main": "dist/main.js",
+  "scripts": {
+    "build": "webpack --display-error-details",
+    "clean": "rm -rf  ./dist/*",
+    "test": "jest --watch",
+    "watch": "webpack",
+    "test:once": "jest"
```

Comment:
Does running jest trigger type checking?

Reply:
No, I don't think so. I'll try to work that in somewhere. When do you think Type checking should happen? On build or on test? both?

Reply:
You know what else I forgot?! **Linting**! I definitely need to get our ESLint config choices integrated in here.

Reply:
Ah right, linting :). Feel free to do a follow-up PR if that's easier. It seems like most people (the internet people) like when the build doesn't fail if type checking fails. As long as it fails in CI, I don't have a preference though.

Reply:
So after going down the rabbit hole a little bit, I found out that indeed Jest **is** doing type checking since we are using the `ts-jest` [transform](https://github.com/cockroachdb/ui/blob/52e2e022d69c39b06f53f805f661f8cc4ab3c888/packages/ui-components/jest.testing.config.js#L12). I am hooking up linting and [type-checking](https://github.com/azz/jest-runner-tsc) anyway, since that seems to check the [definitions file](https://github.com/cockroachdb/ui/blob/52e2e022d69c39b06f53f805f661f8cc4ab3c888/packages/ui-components/src/declaration.d.ts) for issues.

Getting Jest, Webpack, Typescript, Babel, and whoever else to work together seemed hard enough that I don't really want to change anything. So let me know if at some point in the future the type checking gets too cumbersome.